src/Storage/PathBuilder/LocalPathBuilder.php

```diff
@@ -1,11 +1,45 @@
 <?php
 /**
  * @author Florian Krämer
+ * @author Bernhard Picher
  * @copyright 2012 - 2015 Florian Krämer
  * @license MIT
  */
 namespace Burzum\FileStorage\Storage\PathBuilder;
 
+use Burzum\FileStorage\Storage\StorageTrait;
+use Cake\Datasource\EntityInterface;
+
 class LocalPathBuilder extends BasePathBuilder {
 
+	use StorageTrait;
+
+/**
+ * Constructor
+ *
+ * Default options for compatibility reasons:
+ * - pathPrefix = 'files'
+ * - randomPath = 'crc32'
+ *
+ * @param array $config Configuration options.
+ */
+	public function __construct(array $config = []) {
+		$this->_defaultConfig['pathPrefix'] = 'files';
+		$this->_defaultConfig['randomPath'] = 'crc32';
```

Comment:
Actually you're breaking another thing here. crc32 is faulty, if you move from a 32 to a 64 bit system it will create different hashes. This has been fixed in the new paht builder(s).

What you did here is basically changing the whole local path builder back to behave like the old listener, which is not a good idea. This was intentionally done. The path builder already provides BC by the config options.

Please take a look at the [LegacyLocalListener](https://github.com/burzum/cakephp-file-storage/blob/3.0-event-overhaul/src/Storage/Listener/LegacyLocalFileStorageListener.php). Everything you've done is already solved and done through this listener. The plan is, when the old listeners are removed (in ~12 month) to point people to this listener for BC.